src/compute-client/src/controller/replica.rs

```diff
@@ -337,137 +288,5 @@ where
         );
 
         self.metrics.inner.response_queue_size.inc();
-
-        // Apply changes to the `initial_output_duration_seconds` metric.
-        let collection_frontier = match response {
-            ComputeResponse::FrontierUpper { id, upper } => Some((id, upper.clone())),
-            ComputeResponse::SubscribeResponse(id, resp) => match resp {
-                SubscribeResponse::Batch(batch) => Some((id, batch.upper.clone())),
-                SubscribeResponse::DroppedAt(_) => Some((id, Antichain::new())),
-            },
```

Comment:
We used to update the hydration flag in response to these subscribe responses previously, correct? If we only call `update_hydration_status` in response to `handle_frontier_uppers`, how is the same behavior guaranteed now?

Reply:
It's not, good catch!

Reply:
Fixed by calling `update_hydration_status` also in `handle_subscribe_response`. I also added a cluster test to verify that subscribes show up as hydrated in `mz_compute_hydration_statuses`.